TypeScript.py

```diff
@@ -1501,6 +1531,92 @@ def run(self, text):
             formatRange(text, view, rblineStart, ralineEnd)
 
 
+class NavToCommand(sublime_plugin.WindowCommand):
+    navto_panel_started = False
+
+    # indicate if the insert_text command has finished pasting text into the textbox.
+    # during which time the on_modified callback shouldn't do angthing
+    insert_text_finished = False
+
+    # when a new search string comes in, the command state needs to be reset and then
+    # restart. However the reset cannot be reset after the "run" method. Because
+    # the exeuation order of the "run" and "on_done" methods are not fixed, this flag is
+    # set to make sure the state reset to be done before run
```

Comment:
Let's talk in person tomorrow so I can understand what's going on here a little better.